docusaurus/docs/reactnative/guides/performance_boost.mdx

```diff
@@ -0,0 +1,91 @@
+
+- For Livestream type channels, we recommend disabling Connect, Read Events, and Typing Events. These will cause performance issues and don't generally add to the user experience in these use cases.
+- Also consider using [Slow Mode](https://getstream.io/chat/docs/slow_mode/?language=js#channel-slow-mode) for Livestream events.
+
+<img src={channelSettings} />
+
+## State update throttling
+
+[Channel](../core-components/channel.mdx) component updates its internal react state based on chat related events received from websocket.
+By default these state updates are throttled to once per 500 ms. For high traffic applications, you may want to adjust throttling
+interval to a higher number. You can configure throttle interval via props on Channel component:
+
+- `newMessageStateUpdateThrottleInterval` prop adjusts the throttling interval for state updates when new messages arrive in chat.
+- `stateUpdateThrottleInterval` prop adjusts the throttling interval for state updates for all the other updates to channel except new message.
+
+We recommend following configuration, but you can adjust it as per your requirement.
+
+```tsx
+<Channel
+  newMessageStateUpdateThrottleInterval={2000}
+  stateUpdateThrottleInterval={800}
+/>
+```
+
+:::note
+- These props are available in version >= v3.9.0
+- Changes to `stateUpdateThrottleInterval` will result in delays in adding a reaction to message. So try not to keep too high value for this prop.
+:::
+
+## Heavy UI features
+
+You can relieve the JS thread by disabling some of the heavy features as mentioned below.
+
+:::note
+These props are available in version >= v3.9.0
+:::
+
+### Image Viewer
+
+By default, when you open (or press on) an image attachment, it opens the image in full screen image viewer.
+And within the viewer, you can keep swiping left to scroll through all the images within channel (which have been loaded so far).
+This creates quite a lot of extra work for JS thread in terms of keeping track of image attachments loaded in channel and pre-populating them in the viewer
+for smooth transitions.
+
+Since v3.9.0 we have introduced an alternate UX for image viewer, where image attachments of only the selected messages will be loaded in the viewer.
+You can enable this alternate UX by setting a prop `legacyImageViewerSwipeBehaviour` as false on Channel component.
+
+```tsx
+<Channel legacyImageViewerSwipeBehaviour={false} />
+```
+
+### Sticky Date Header and Inline Date Separator
+
+By default MessageList component renders sticky date header a the top, and date separators in between messages from different dates.
+Computation and insertion of these dates can be heavy for message list when there are lot of messages in channel, since it involves iterating
+through the entire list. For livestream applications we would recommend disabling these features using following two props on Channel component.
```

Comment:
```suggestion
By default, the MessageList component renders a sticky date header at the top of the list, and date separators between messages from different dates.
Computation and insertion of these dates can be heavy for a message list when there are a lot of messages in the channel since it involves iterating through the entire list. 

For livestream applications, we would recommend disabling these features using the following two props on a Channel component:
```